src/sentry/replays/usecases/query/__init__.py

```diff
@@ -163,6 +165,20 @@ def query_using_optimized_search(
             SearchFilter(SearchKey("environment"), "IN", SearchValue(environments)),
         ]
 
+    for i in range(len(search_filters)):
+        sf = search_filters[i]
+        if sf.key.name in VIEWED_BY_ME_KEY_ALIASES:
+            # "viewed_by_me" is not a valid query field.
+            # It's a convenience alias for users without the admin access to lookup ids.
+            if sf.operator not in ["=", "!="]:
+                raise ParseError(f"Invalid operator specified for {sf.key.name}")
+            is_me = not (
+                (sf.operator == "=") ^ sf.value.value
+            )  # True if both expressions are the same, else False.
```

Comment:
Very fancy!  Let's change this though.  Only the `=` operator should be allowed.  I think `!=` doesn't make sense in this context.  Now we only need to care about the value and can drop the xor.

Reply:
The problem is the frontend search bar: users don't type in `key=val` or `key!=val`, they type `key:val` and `!key:val`. The latter translates into the != operator. So unless we want to drop searches for `!viewed_by_me:x`, we have to handle !=...

Reply:
I guess if users know its a boolean, it'd be pretty weird to do `!viewed_by_me:true`. So we can just raise ParseError to simplify this..